src/main/java/net/eno/chess/Board.java

```diff
@@ -1,87 +1,115 @@
 package net.eno.chess;
 
-import net.eno.pieces.Piece;
 import net.eno.pieces.Color;
+import net.eno.pieces.Piece;
 import net.eno.pieces.PieceType;
+import net.eno.pieces.Position;
+
 import static net.eno.utils.StringUtils.appendNewLine;
 
+import java.util.Collections;
 import java.util.List;
 import java.util.ArrayList;
 
 
 public class Board {
 
-    private final List<List<Piece>> board;
+    private List<Rank> board;
 
     public Board() {
         this.board = new ArrayList<>();
-        for (int i = 0; i < 4; i++) {
-            this.board.add(new ArrayList<>());
-        }
     }
 
-    public void addPiece(int rank, Piece piece) {
-        this.board.get(rank).add(piece);
+    private void addRank(Rank rank) {
+        this.board.add(rank);
     }
 
-    public int pieceCount() {
-        int pieceCount = 0;
-        for (List<Piece> rank : this.board) {
-            pieceCount += rank.size();
+    public int countTargetPiece(Color color, PieceType pieceType) {
+        int count = 0;
+        for (Rank rank : this.board) {
+            count += rank.countTargetPiece(color, pieceType);
         }
-        return pieceCount;
+        return count;
     }
 
-    public void initialize() {
-        List<PieceType> arrangePieceList = arrangePiece();
-        for (int i = 0; i < 8; i++) {
-            addPiece(0 , Piece.createPiece(Color.BLACK, arrangePieceList.get(i)));
-            addPiece(1, Piece.createPiece(Color.BLACK, PieceType.PAWN));
-            addPiece(2, Piece.createPiece(Color.WHITE, PieceType.PAWN));
-            addPiece(3, Piece.createPiece(Color.WHITE, arrangePieceList.get(i)));
+    public Piece findPiece(String position) {
+        Position positionObj = new Position(position);
+        Rank targetRank = this.board.get(positionObj.getRankIndex());
+        return targetRank.findPiece(positionObj.getFileIndex());
+    }
+
+    public void move(String position, Piece piece) {
+        Position positionObj = new Position(position);
+        Rank targetRank = this.board.get(positionObj.getRankIndex());
+        targetRank.move(positionObj.getFileIndex(), piece);
+    }
+
+    public double calculatePoint(Color color) {
+        double point = 0;
+        for (Rank rank : this.board) {
+            point += rank.calculateRankPoint(color);
+        }
```

Comment:
`stream` 을 이용해서 개선이 가능해보입니다. 

```suggestion
        double point = this.board.stream().mapToDouble(rank.calculateRankPoint(color)).sum()

Reply:
전체적으로 for문을 사용한 부분을 가능한한 stream을 이용해서 리팩토링 해봤습니다.
어떤 부분은 가독성이 좋아진 반면 오히려 더 복잡해진거 아닌가?? 싶은 부분도 있었습니다.
Board class에 `sortByPiecePoint` 메소드나 Board class에 `showBoard` 메소드가 그런 것 같아요.
`sortByPiecePoint`는 체이닝이 4번이나 있고 `showBoard`같은 경우는 result를 만드는 것 까진 문제가 없지만 return 할 때 검은색인 경우 테스트 케이스를 통과하기 위해 subString 메소드와 NEWLINE을 붙이는 수고를 들이고 있습니다. (reverse로 뒤집으면 줄바꿈도 뒤집히더라구요.)

그리고 Board class에 `initializeEmpty` 메소드와 Rank class에 `createOnePieceRank` 메소드만 for문을 없애지 않았습니다. 이 부분은 아무리 생각해도 for문보다 나은 방법이 생각나질 않더라구요.
이 부분에 대해서도 의견 부탁드릴게요!!

위에 언급한 메소드나 그 외에 메소드까지 비효율적인 부분이나 개선하면 좋은 점이 있다면 말씀 부탁드리겠습니다!!